src/libcephfs.cc

```diff
@@ -2019,31 +2019,132 @@ extern "C" int64_t ceph_ll_writev(class ceph_mount_info *cmount,
   return (cmount->get_client()->ll_writev(fh, iov, iovcnt, off));
 }
 
+struct ceph_ll_readv_writev_buffer {
+  ceph_ll_readv_writev_buffer() : iov() {};
+  ~ceph_ll_readv_writev_buffer() {
+    delete iov;
+  }
+  void prepare_iovs() {
+    ceph_assert(bl.get_num_buffers() <= IOV_MAX);
```

Comment:
This could be an assert, but we'd have to gate on it by every user invoking it — which doesn't happen that I can see. So it either needs to be allowed to return an error code, and do so (if that makes any sense), or else you need to add the appropriate gating on all paths in.

Reply:
Frank pointed out to me elsewhere that I misread things and this isn't user input — it wants to marshal the bufferlist of data returned by the messenger on message reads, and share it directly.
I'm...not sure how I feel about giving this memory out to our users directly. Investigating.

Reply:
I am reasonably sure (though not certain) that you can expect the number of buffers in this direct-read-from-OSDs scenario to be the number of objects that were read, as it will be the data gathered from each of the incoming Message decodings. So this is proooooobably safe, and it's not particularly special memory buffers that we need to quickly deallocate.

Reply:
OK, I believe I have eliminated all the asserts I added.